drivers/flash/flash_stm32.h

```diff
@@ -53,6 +53,19 @@ struct flash_stm32_priv {
 
 
 /* Redefinitions of flags and masks to harmonize stm32 series: */
+#if defined(CONFIG_SOC_SERIES_STM32U5X)
+#define FLASH_NSCR_NSLOCK FLASH_NSCR_LOCK
+#define FLASH_OPTR_DBANK FLASH_OPTR_DUALBANK
+#define FLASH_NSCR_NSPG FLASH_NSCR_PG
+#define FLASH_NSCR_NSBKER_Msk FLASH_NSCR_BKER_Msk
+#define FLASH_NSCR_NSBKER FLASH_NSCR_BKER
+#define FLASH_NSCR_NSPER FLASH_NSCR_PER
+#define FLASH_NSCR_NSPNB_Msk FLASH_NSCR_PNB_Msk
+#define FLASH_NSCR_NSPNB_Pos FLASH_NSCR_PNB_Pos
+#define FLASH_NSCR_NSPNB FLASH_NSCR_PNB
+#define FLASH_NSCR_NSSTRT FLASH_NSCR_STRT
+#endif /* CONFIG_SOC_SERIES_STM32U5X */
+
```

Comment:
For clarity, if we need to provide zephyr specific abstraction on top of Cube apis, avoid re-using Cube definition and create zephyr specific ones.

Reply:
the fact is that FLASH_NSCR_NSLOCK is in the flash_stm32.c or FLASH_NSCR_NSPNB_Msk in the flash_stm32l5x.c etc

Reply:
Yes

Reply:
The FLASH_NSCR_LOCK is the name of the bit registered in the stm32CUbe/stm32l5x HAL
Changing FLASH_NSCR_LOCK to FLASH_STM32_NSCR_LOCK in the zephyr driver means 
  a change in the drivers/flash/stm32_flash.h and also updating drivers/flash/flash_stm32l5x.c, drivers/flash/flash_stm32g4x.c and drivers/flash/flash_stm32l4x

Reply:
cf commit 4a94eedbc1a428f1013a662a318022a6d5499eb9

Reply:
I agree this is not straightforward but we should keep each level of abstraction clear. Otherwise this will lead to confusion in the end (if this is spreading in other drivers)
Alternatives are:
- get CMSIS definitions fixed (but LL is here to provide CMSIS abstraction)
- convert whole driver to use of LL